joblib/test/test_my_exceptions.py

```diff
@@ -2,6 +2,7 @@
 Test my automatically generate exceptions
 """
 from joblib import my_exceptions
+from _pytest.assertion.util import BuiltinAssertionError
```

Comment:
Yes, it raised an `AssertionError` as `JoblibAssertionError` is not an instance of `_pytest.assertion.util.AssertionError`. While running the test with pytest, we have to reference the original `AssertionError` shipped by core python like this.